zerver/actions/bots.py

```diff
@@ -84,9 +146,22 @@ def do_change_bot_owner(
             event_type=RealmAuditLog.USER_BOT_OWNER_CHANGED,
             event_time=event_time,
         )
+        removed_subs_list = remove_bot_from_inaccessible_private_streams(
+            user_profile, acting_user=acting_user
+        )
 
     send_bot_owner_update_events(user_profile, bot_owner, previous_owner)
 
+    unsubscribed_streams = [stream for (user_profile, stream) in removed_subs_list]
+    send_subscription_remove_events(
+        user_profile.realm, [user_profile], unsubscribed_streams, removed_subs_list
+    )
+
+    if unsubscribed_streams:
+        deactivate_private_streams_after_unsubcribtions(
+            user_profile.realm, unsubscribed_streams, acting_user=acting_user
+        )
```

Comment:
Have kept this at last like we do in `bulk_remove_subscriptions`, i.e. deactivating streams after doing all the event related work.

Reply:
`unsubcribtions` has a typo :)

Reply:
It was present before, when I commented the above statement. Was fixed later.